core/templates/dev/head/pages/topic_editor/TopicEditorStateService.js

```diff
@@ -209,11 +212,43 @@ oppia.factory('TopicEditorStateService', [
        * _subtopicPages list.
        */
       setSubtopicPage: function(subtopicPage) {
-        if (_getSubtopicPageIndex(subtopicPage.getId()) !== -1) {
+        if (_getSubtopicPageIndex(subtopicPage.getId()) !== null) {
           _subtopicPages[_getSubtopicPageIndex(subtopicPage.getId())] =
             angular.copy(subtopicPage);
         } else {
           _setSubtopicPage(subtopicPage);
+          _newSubtopicPages.push(angular.copy(subtopicPage));
+        }
+      },
+
+      deleteSubtopicPage: function(topicId, subtopicId) {
+        var subtopicPageId = topicId + '-' + subtopicId.toString();
```

Comment:
This logic comes up several times. We should encapsulate it in a common function throughout the frontend so that there is a single source of truth.

Reply:
Done.